examples/siamese/create_db.py

```diff
@@ -0,0 +1,253 @@
+VAL_IMAGE_COUNT = 1000
+
+
+def create_lmdbs(folder, file_list, image_width=None, image_height=None, image_count=None):
+    """
+    Creates LMDBs for generic inference
+    Returns the filename for a test image
+
+    Creates these files in "folder":
+        train_images/
+        train_labels/
+        val_images/
+        val_labels/
+        mean.binaryproto
+        test.png
+    """
+    if image_width is None:
+        image_width = IMAGE_SIZE
+    if image_height is None:
+        image_height = IMAGE_SIZE
+
+    if image_count is None:
+        train_image_count = TRAIN_IMAGE_COUNT
+    else:
+        train_image_count = image_count
+    val_image_count = VAL_IMAGE_COUNT
+
+    # read file list
+    images = []
+    f = open(file_list)
+    for line in f.readlines():
+        line = line.strip()
+        if not line:
+            continue
+
+        path = None
+        # might contain a numerical label at the end
+        match = re.match(r'(.*\S)\s+(\d+)$', line)
+        if match:
+            path = match.group(1)
+            ground_truth = int(match.group(2))
+            images.append([path,ground_truth])
+
+    print "Found %d image paths in image list" % len(images)
+
+    for phase, image_count in [
+            ('train', train_image_count),
+            ('val', val_image_count)]:
+
+        print "Will create %d pairs of %s images" % (image_count, phase)
+
+        # create DBs
+        image_db = lmdb.open(os.path.join(folder, '%s_images' % phase),
+                map_async=True,
+                max_dbs=0)
+        label_db = lmdb.open(os.path.join(folder, '%s_labels' % phase),
+                map_async=True,
+                max_dbs=0)
+
+        # add up all images to later create mean image
+        image_sum = None
+        shape = None
+
+        # save test images (one for each label)
+        testImageSameClass = None
+        testImageDifferentClass = None
+
+        for i in xrange(image_count):
+            # pick up random indices from image list
+            index1 = random.randint(0, len(images)-1)
+            index2 = random.randint(0, len(images)-1)
+            # label=1 if images are from the same class otherwise label=0
+            label = 1 if int(images[index1][1]) == int(images[index2][1]) else 0
+            # load images from files
+            image1 = np.array(utils.image.load_image(images[index1][0]))
+            image2 = np.array(utils.image.load_image(images[index2][0]))
+            if not shape:
+               # initialize image sum for mean image
+               shape = image1.shape
+               image_sum = np.zeros((3,shape[0],shape[1]), 'float64')
+            assert(image1.shape == shape and image2.shape == shape)
+
+            # create BGR image: blue channel will contain first image,
+            # green channel will contain second image
+            image_pair = np.zeros(image_sum.shape)
+            image_pair[0] = image1
+            image_pair[1] = image2
```

Comment:
After discussion, I actually like having a 3-channel image here. It's easier to work with images in a standard format. Plus, showing off the `Silence` layer in the network is neat.